packages/rpc-subscriptions-spec/src/rpc-subscriptions-channel.ts

```diff
@@ -26,3 +26,36 @@ export type RpcSubscriptionsChannelCreator<TOutboundMessage, TInboundMessage> =
         abortSignal: AbortSignal;
     }>,
 ) => Promise<RpcSubscriptionsChannel<TOutboundMessage, TInboundMessage>>;
+
+export function transformChannelInboundMessages<TOutboundMessage, TNewInboundMessage, TInboundMessage>(
+    channel: RpcSubscriptionsChannel<TOutboundMessage, TInboundMessage>,
+    transform: (message: TInboundMessage) => TNewInboundMessage,
+): RpcSubscriptionsChannel<TOutboundMessage, TNewInboundMessage> {
+    return Object.freeze<RpcSubscriptionsChannel<TOutboundMessage, TNewInboundMessage>>({
+        ...channel,
+        on(type, subscriber, options) {
+            if (type !== 'message') {
+                return channel.on(
+                    type,
+                    subscriber as (data: RpcSubscriptionChannelEvents<TInboundMessage>[typeof type]) => void,
+                    options,
+                );
+            }
+            return channel.on(
+                'message',
+                message => (subscriber as (data: TNewInboundMessage) => void)(transform(message)),
+                options,
+            );
+        },
+    });
+}
+
+export function transformChannelOutboundMessages<TNewOutboundMessage, TOutboundMessage, TInboundMessage>(
+    channel: RpcSubscriptionsChannel<TOutboundMessage, TInboundMessage>,
+    transform: (message: TNewOutboundMessage) => TOutboundMessage,
```

Comment:
I think this type is backward, unless I'm mis-understanding. The channel currently produces `TOutboundMessage` so I think we want the transform to be `TOutboundMessage => TNewOutboundMessage`.

Reply:
Hmm no I think that's right. Consider the following example:

![image](https://github.com/user-attachments/assets/873ca490-e046-402e-9847-7ee826344157)

If I have a channel that sends `X` messages and I want to send `T` messages, I need `T => X` not `X => T`.

Reply:
Ah okay yep you're doing the modification the other way to what I thought. I thought you were writing a transformed channel that sits between the inner channel and the server. Gotcha, thanks!